src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -285,24 +285,27 @@ bool QMCDriverNew::finalize(int block, bool dumpwalkers)
   return true;
 }
 
-void QMCDriverNew::makeLocalWalkers(IndexType nwalkers,
+void QMCDriverNew::makeLocalWalkers(IndexType nwalkers_global,
+                                    IndexType nwalkers_local,
                                     RealType reserve,
                                     const ParticleAttrib<TinyVector<QMCTraits::RealType, 3>>& positions)
 {
+  population_.set_num_global_walkers(nwalkers_global);
```

Comment:
Perhaps there was a sync across ranks for establishing the global population in the usual way that was dropped from VMCBatch? If we assume all went well with the makeLocal we should call population_.set_num_global_walkers in QMCDriverNew::startup. It's certainly a startup only bit of logic.

Reply:
#2575 deleted lines setting global walkers and introduced a bug.
I think when the local walkers are touched, the num_global_walkers must be updated. These two should be put as close as possbile.
This logic should be applied by startup and load balance.

Isn't makeLocal start up (per driver) only?

Reply:
This is the preferred way to set the global walkers, but considering the name it still feels like a side effect.
`population_.syncWalkersPerNode(myComm);`

I still prefer the makeLocal not to do anything but makeLocal.
This is the exception in that multiple conditional adjustments aren't made to the local population. I do these and when finished call syncWalkersPerNode. When to do this is best decided in that scope rather than by figuring out what local walker count effecting methods should call sync and which shouldn't.

This should probably be
`IndexType syncNumGlobalWalkers(myComm)`
which updates num_global_walkers_;

While `get_num_global_walkers` is still the accessor to the possibly stale member variable left for optimization.
I will update documentation in the next PR that touches population.

Reply:
@PDoakORNL and I had a call. syncWalkersPerNode() will be probably not needed overall. The PR has been updated.